tests/Integration/PrestaShopBundle/Controller/Sell/Customer/Address/AddressControllerTest.php

```diff
@@ -0,0 +1,249 @@
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\PrestaShopBundle\Controller\Sell\Customer\Address;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\BrowserKit\Client;
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Component\DomCrawler\Form;
+use Tests\Integration\PrestaShopBundle\Controller\FormFiller\FormFiller;
+
+class AddressControllerTest extends WebTestCase
+{
+    /**
+     * This will be modified during the setUp
+     *
+     * @var int
+     */
+    private $testAddressId;
+
+    /**
+     * @var FormFiller
+     */
+    private $formFiller;
+
+    public function __construct($name = null, array $data = [], $dataName = '')
+    {
+        parent::__construct($name, $data, $dataName);
+
+        $this->formFiller = new FormFiller();
+    }
+
+    /**
+     * Creates a test address and ensures that there are 3 addresses so filtered values can be counted correctly.
+     */
+    public function setUp()
+    {
+        $client = static::createClient();
+        $this->createTestAddress($client);
+        $router = $client->getKernel()->getContainer()->get('router');
+        $addressUrl = $router->generate('admin_addresses_index');
+        $crawler = $client->request('GET', $addressUrl);
+        $addresses = $this->getAddressList($crawler);
+        self::assertEquals(3, count($addresses));
+        $this->validateTestAddressExists($addresses);
+    }
+
+    /**
+     * Removes the created test address and ensures that it's successfully removed from the list.
+     */
+    public function tearDown()
+    {
+        $client = static::createClient();
+        $client->followRedirects(true);
+        $router = $client->getKernel()->getContainer()->get('router');
+        $addressDeleteUrl = $router->generate('admin_addresses_delete', ['addressId' => $this->testAddressId]);
+        $crawler = $client->request('POST', $addressDeleteUrl);
```

Comment:
What if we have no eligible controller to delete the data 😛  (like a Stock movement, I'm not sure it can be deleted).

I would suggest then
- either a database reset (can be costly)
- or you can bypass the controller level and directly use the under layer (like call `$objectModel->delete(...)`)

Reply:
Good idea regarding object layer. But I think it makes sense to use routes to delete where possible, it also serves as "soft" test for deletion working.